### PR TITLE
Fixes for plugin progress dialog cancel & LR lockup.

### PR DESCRIPTION
**Pull request by philmoz**
This change allows the progress dialog to be cancelled.
It also prevents the plugin from being run twice which locks up LR with a 'Task Completed' dialog box that can't be closed.

### CODE REVIEW

**Comment by musselwhizzle** [comment]
how does it prevent it from running twice? Does startAsyncTask use single thread pool or something?

**Comment by philmoz** [comment]
It appears that doing the call to 'checkPhotoAvailability' outside the 'innerContext' was allowing you to trigger the plugin twice. I was only able to do this by mapping a keyboard shortcut to the plugin - sometimes the plugin would take a while to start and I would hit the shortcut again.

I noticed that the metadata dialog plugin didn't do this, and the main difference was the extra code being run before the 'innerContext'.

File: focuspoints.lrdevplugin/FocusPoint.lua [hunk 1]
```diff
@@ -14,27 +14,37 @@
   limitations under the License.
 --]]
 
-local LrSystemInfo = import 'LrSystemInfo'
 local LrFunctionContext = import 'LrFunctionContext'
 local LrApplication = import 'LrApplication'
 local LrDialogs = import 'LrDialogs'
-local LrView = import 'LrView'
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/FocusPoint.lua]: FYI, this is the exact reason I suggest the AppleDelegates require it's own imports. With this refactoring, it's possible it broke other delegates.

**Comment by musselwhizzle** [comment]
> It appears that doing the call to 'checkPhotoAvailability' outside the 'innerContext' was allowing you to trigger the plugin twice.

gezzz......

File: focuspoints.lrdevplugin/FocusPoint.lua (hunk 1 above)
Comment (philmoz) [inline focuspoints.lrdevplugin/FocusPoint.lua]: Which delegates does it break? 
I tried it on a variety of test images I have and could not find any issues.

**Comment by musselwhizzle** [comment]
turns out the dialog blocks all UI input which is what stops it from opening twice. Move the photo check to inside past the progress dialog opening is the trick needed.

**Comment by musselwhizzle** [comment]
thanks combined the code and merged in.

**Pull request merged**